libvirt/resource_libvirt_domain.go

```diff
@@ -699,6 +699,25 @@ func resourceLibvirtDomainRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Error retrieving interface addresses: %s", err)
 	}
 
+	// compare number of interfaces in ifacesWithAddr with number of interfaces in domainXML
+	// if they're not equal, get the ifacesWithAddr structure from the network xmls
+	interfacesXML := make(map[string]bool)
+	interfacesIfaces := make(map[string]bool)
+
+	for _, networkInterface := range domainDef.Devices.NetworkInterfaces {
+		interfacesXML[networkInterface.Source.Network] = true
+	}
+	for _, virDom := range ifacesWithAddr {
+		interfacesIfaces[virDom.Name] = true
+	}
+
+	if len(interfacesXML) > len(interfacesIfaces) {
+		ifacesWithAddr, err = getDomainInterfacesFromNetworks(domainDef, meta)
+		if err != nil {
+			return fmt.Errorf("Error retrieving interface addresses: %s", err)
+		}
+	}
+
```

Comment:
Maybe we could move this whole block [inside `getDomainInterfaces`](https://github.com/dmacvicar/terraform-provider-libvirt/blob/master/libvirt/resource_libvirt_domain.go#L979)...